### ISSUE TITLE
Tagged release 1.3.0 not available

### ISSUE THREAD

**Issue by joancafom** [Kong/kubernetes-ingress-controller#1379]
### Summary

Hi! 

It seems that the latest tagged release [1.3.0](https://github.com/Kong/kubernetes-ingress-controller/releases/tag/1.3.0) does include the changes for [2.0.0-alpha1](https://github.com/Kong/kubernetes-ingress-controller/releases/tag/2.0.0-alpha.1) instead:

![Screenshot 2021-05-28 at 12 48 09](https://user-images.githubusercontent.com/24253844/119972817-f6c32480-bfb2-11eb-9717-19ca2d1f1685.png)

**Kong Ingress controller version**
1.3.0

**Kong or Kong Enterprise version**
NA

### Kubernetes version
NA

### Environment

- **Cloud provider or hardware configuration**: NA
- **OS** (e.g. from /etc/os-release): NA
- **Kernel** (e.g. `uname -a`): NA
- **Install tools**: NA
- **Others**: NA

### Steps To Reproduce

```bash
$ wget https://github.com/Kong/kubernetes-ingress-controller/archive/refs/tags/1.3.0.tar.gz
$ tar xfz 1.3.0.tar.gz
$  cat kubernetes-ingress-controller-1.3.0/Makefile | head -n1
TAG?=2.0.0-alpha.1
```

**Comment by shaneutt** [comment]
Thanks for this report, we're going to fix this and patch to `v1.3.1` to get it taken care of I'll report back when that's done.

**Comment by shaneutt** [comment]
Thanks for this report. The changes for both `v1.3.0` and `v2.x` live in the same branch under different directories and so you will see some of that code (under `railgun/`) present in the REF, _however_ there's different builds and artifacts for `v1.x` vs `v2.x`, so just note the _inclusion of both in the code is expected_.

That said, having the new default tag be the alpha one doesn't work well as a default, we're going to fix this and patch to `v1.3.1` with that changed.

In the meantime for `v1.3.0` you can manually override the build with the correct tag in this manner:

```shell
docker run -it kong/kubernetes-ingress-controller:1.3.0
```

**Comment by shaneutt** [comment]
(just a note: I deleted my previous response as there was some incorrect information in there, new info is incoming)

**Comment by mflendrich** [comment]
Hi @joancafom,

The v2.x line is being developed as a separate executable in the same repository. The entry point for v1.x is in the `/cli/ingress-controller/main.go` file while for v2.x it's in the `/railgun/main.go` file.

We've been developing both in parallel, and both the `v1.3` and the `v2.0.0-alpha.1` are happening simultaneously, but from two different artifacts compiled from the same commit. Hence no expected diff between these two tags. However, when interpreting these two tags, the reader needs to know that they can build `v1.x` from the former location and `v2.x` from the latter.

The source of confusion comes from the fact that there is only one `TAG` variable in the Makefile despite two different versions being buildable from this commit. We've created #1381 to address that problem.

If you pull the `kong/kubernetes-ingress-controller:1.3.0` tag, you'll get the valid 1.3.0 image. Similarly, if you pull the `kong/kubernetes-ingress-controller:v2.0.0-alpha.1` tag, you'll get the 2.x artifact. They happen to be built from the same commit, but the former is from `/Dockerfile` and the latter is from `/Dockerfile.railgun`.

The 1.3.0 image is correct, but some users building 1.3.0 from source may have it accidentally advertise itself as 2.0.0-alpha.1 due to no distinction between tags for these two artifacts. To solve that problem, the aforementioned #1381 will make these separate.

In short:
- it's safe to use the `kong/kubernetes-ingress-controller:1.3.0` image - it contains the right `1.3.0` build
- it's safe to use the `kong/kubernetes-ingress-controller:2.0.0-alpha.1` image - it contains the right `2.0.0-alpha.1` build (but remember it's alpha quality now)
- those who've tried building `1.3.0` from source got the `1.3.0` build that advertises itself as `2.0.0-alpha.1`. #1381 is about to fix that. The `1.3.1` which we'll release today won't have that problem anymore.

Thanks for your report and sorry about the confusion from our side.
